File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-fortran/tsvc_2_s316__submitted.f90
subroutine tsvc_2_s316_fp64(a, result, LEN_1D, workspace, workspace_bytes) bind(C)
    use iso_c_binding
    implicit none
    integer(c_int64_t), value :: LEN_1D
    type(c_ptr), value :: workspace
    integer(c_int64_t), value :: workspace_bytes
    real(c_double), intent(in) :: a(*)
    real(c_double), intent(out) :: result(1)
    integer(c_int64_t) :: i
    real(c_double) :: x

    if (LEN_1D <= 0_c_int64_t) then
        result(1) = 0.0_c_double
        return
    end if

    x = a(1)

    !$omp parallel do reduction(min:x) private(i) schedule(static)
    do i = 2, LEN_1D
        if (a(i) < x) x = a(i)
    end do
    !$omp end parallel do

    result(1) = x
end subroutine tsvc_2_s316_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s316__submitted.c
#include <stdint.h>
#include <stdio.h>
#include <math.h>
#include <omp.h>

void tsvc_2_s316_fp64(double const* restrict a, double* restrict result, int64_t LEN_1D, uint8_t* restrict workspace, int64_t workspace_size) {
    (void)workspace;
    (void)workspace_size;
    double x = a[0];
    #pragma omp parallel for reduction(min:x) schedule(static)
    for (int64_t i = 1; i < LEN_1D; ++i) {
        if (a[i] < x) {
            x = a[i];
        }
    }
    result[0] = x;
}
